DataGeneration/dataGen.js

```diff
@@ -0,0 +1,88 @@
+const faker = require('faker');
+
+const createCsvStringifier = require('csv-writer').createObjectCsvStringifier;
+const csvStringifier = createCsvStringifier({
+    header: [
+        {id: 'id', title: 'ID'},
+        {id: 'title', title: 'TITLE'},
+        {id: 'description', title: 'DESC'},
+        {id: 'originalPrice', title: 'OGPRICE'},
+        {id: 'salePrice', title: 'SALEPRICE'},
+        {id: 'colors', title: 'COLORS'},
+        {id: 'sizes', title: 'SIZES'},
+        {id: 'liked', title: 'LIKED'},
+        {id: 'inStock', title: 'INSTOCK'},
+        {id: 'reviews', title: 'REVIEWS'},
+    ]
+});
+var fs = require('fs');
+
+
+
+var stream = fs.createWriteStream('DataGeneration/data.csv')
+var max = 10000000
+var i = 0;
+
+
+const createFakeData = () => {
+  var ok = true;
+
+  do {
+    let price = faker.commerce.price();
+    let reviews = [];
+    let colors = [];
+    let sizes = [];
+
+    for (let j = 0; j < Math.floor(Math.random() * 8); j++) {
+      let easeOfAssembly = Math.floor(Math.random() * 5 + 1);
+      let valueForMoney = Math.floor(Math.random() * 5 + 1);
+      let productQuality = Math.floor(Math.random() * 5 + 1);
+      let appearance = Math.floor(Math.random() * 5 + 1);
+      let worksAsExpected = Math.floor(Math.random() * 5 + 1);
+      let overallRating = Math.floor((easeOfAssembly + valueForMoney + productQuality + appearance + worksAsExpected) / 5);
+      let createdAt = faker.date.past();
+      let iRecommendThisProduct = faker.random.boolean();
+      let header = faker.lorem.words();
+      let body = faker.lorem.paragraphs()
+
+      reviews.push({ overallRating, easeOfAssembly, valueForMoney, productQuality, appearance, worksAsExpected, createdAt, iRecommendThisProduct, header, body})
+    }
+
+    for (let j = 0; j < Math.floor(Math.random() * 6); j++) {
+      colors.push(faker.commerce.color())
+    }
+
+    let sizeOptions = ['S', 'M', 'L', 'XL']
+    for (let j = 0; j < Math.floor(Math.random() * 4); j++) {
+      sizes.push(sizeOptions[j]);
+    }
+
+    let dataObj = {
+      id: i,
+      title: faker.lorem.word(),
+      description: faker.lorem.sentence(),
+      originalPrice: price,
+      salePrice: price * 0.85,
+      colors: colors,
+      sizes: sizes,
+      liked: false,
+      inStock: Math.random() <= 0.2 ? 0 : Math.floor(Math.random() * 15000),
+      reviews: JSON.stringify(reviews)
+    }
+
+    if (i === max) {
+      stream.write(csvStringifier.stringifyRecords([dataObj]));

```

Comment:
Question: Did stringifying the records make the csv file creation faster? I was thinking of doing this but I wasn't how much of a difference it would make.

Reply:
@javsflores It's hard to say. I went through a lot of GB of csv files trying methods to write a good one with 10,000,000 records and stringifying actually seemed to be a little slower given the same amount of data. But it was also the best way that I was able to implement 'drain'. I know you did it differently though, so it would be interesting to see how the methods compare.